tests/qir_test.py

```diff
@@ -240,23 +240,32 @@ def test_single_conditional(self, one_conditional_circuit: Circuit) -> None:
 
         circuit = circuit_from_qir(one_conditional_bc_path)
 
-        non_condition_circuit = Circuit(5, 13)
+        non_condition_circuit = Circuit(circuit.n_qubits, circuit.n_bits)
         for com in circuit.get_commands()[:15]:
             non_condition_circuit.add_gate(com.op, com.args)
 
-        exp_non_condition_circuit = Circuit(5, 13)
+        exp_non_condition_circuit = Circuit(circuit.n_qubits, circuit.n_bits)
         for com in one_conditional_circuit.get_commands()[:15]:
             exp_non_condition_circuit.add_gate(com.op, com.args)
 
         assert non_condition_circuit == exp_non_condition_circuit
 
         condition_com = circuit.get_commands()[15]
+
+        inverse_reg_map = {v: k for k, v in circuit._reg_map.items()}
+        assert inverse_reg_map[condition_com.args[0]].reg_name == "%0"
+
         condition_circuit = condition_com.op.op.get_circuit()
 
         exp_condition_com = one_conditional_circuit.get_commands()[15]
         exp_condition_circuit = exp_condition_com.op.op.get_circuit()
 
-        assert condition_circuit == exp_condition_circuit
+        # Comparing commands as I get a type error while comparing circuits.
+        # TypeError: unhashable type: 'instancemethod'.
```

Comment:
Do you know why? Is this a bug in pytket?

Reply:
Not too sure. If I run the assertion on circuit equivalence, I have got the following error:

```sh
>       assert condition_circuit == exp_condition_circuit
E       AssertionError: assert [Reset q[0]; ... Reset q[0]; ] == [Reset q[0]; ... Reset q[0]; ]
E         (pytest_assertion plugin: representation of details failed: /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/pprint.py:176: TypeError: unhashable type: 'instancemethod'.
E          Probably an object has a faulty __repr__.)

qir_test.py:269: AssertionError
```
However,  the assertion works in the debugger and returns `False` (although circuits are seemingly identical):

```sh
(Pdb) assert condition_circuit == exp_condition_circuit
*** AssertionError
(Pdb) condition_circuit == exp_condition_circuit
False
(Pdb)
```
But it succeeds if I compare commands explicitly.